### ISSUE TITLE
[50.0.0-SNAPSOT] Units doubling up in unit cache

### ISSUE THREAD

**Issue by mhjacks** [MegaMek/mekhq#4604]
### Environment
```
megamek.MegaMek.initializeLogging(MegaMek.java:143) - Starting MegaMek v0.50.0-SNAPSHOT
    Build Date: 2024-08-03T01:16:41.243976560
    Today: 2024-08-03
    Origin Project: MegaMek
    Java Vendor: Red Hat, Inc.
    Java Version: 21.0.4
    Platform: Linux 6.9.12-200.fc40.x86_64 (amd64)
    System Locale: en_US
    Total memory available to MegaMek: 4 GB
    MM Code Revision: 71bd1ef7166a07f477da27334ae77fc15acd42b7
    MML Code Revision: 35aa5ce66ca59272c0d1e05c3d174cd4c0335b3c
    MHQ Code Revision: 76e9cea125956ab9a9d8af389ee60c2bb2eca81d
```

### Description
Units show up twice in unit cache. This only seems to be true of the units that ship with the game (as opposed to customs units I've added). For example the Awesome 11H is doubled:
![image](https://github.com/user-attachments/assets/0d37a666-63b2-4640-9abb-f8f9c43c4e03)

My custom SnH is not:
![image](https://github.com/user-attachments/assets/b733ed68-9024-4ca1-9c91-cb753575b75f)

Every canon unit I've spot checked (including Aerospace fighters, dropships, infantry, tanks, VTOLs) seems to suffer from this.

**Comment by rjhancock** [comment]
Moved to MekHQ as this is where the issue is.

Should be fixed in next nightly. Is fixed in master.

**Comment by mhjacks** [comment]
Confirmed fixed in MekHQ commit a16a4fbdef
